### PR TITLE
Check CI result on cargo make test for available fuzzers

### LINKED ISSUE #1037: Check fuzzing results in CI
In CI, we test each fuzzer by running the fuzzer with a timeout. (`cargo make test`)
This checks if the fuzzer builds and runs without any error.

We should check if the fuzzer has found any new testcases too.

### PR DESCRIPTION
**Pull request by SpaceWhite**
List of available fuzzers with Makefile.toml :
- [x] ./libfuzzer_libpng_cmin/Makefile.toml
- [x] ./libfuzzer_libpng_ctx/Makefile.toml
- [x] ./nautilus_sync/Makefile.toml
- [x] ./libfuzzer_libpng_accounting/Makefile.toml
- [x] ./libfuzzer_libpng/Makefile.toml
- [x] ./libfuzzer_libmozjpeg/Makefile.toml
- [ ] ~./nyx_libxml2_parallel/Makefile.toml~ (Doesn't have tasks.test)
- [x] ./libfuzzer_stb_image/Makefile.toml
- [x] ./fuzzbench_qemu/Makefile.toml
- [ ] ~./forkserver_libafl_cc/Makefile.toml~ (Doesn't have tasks.test)
- [ ] ~./libfuzzer_windows_asan/Makefile.toml~ (tasks.test is empty)
- [ ] ~./frida_gdiplus/Makefile.toml~ (Windows)
- [x] ./libfuzzer_stb_image_sugar/Makefile.toml
- [x] ./libfuzzer_libpng_launcher/Makefile.toml
- [ ] ~./tinyinst_simple/Makefile.toml~ (Windows)
- [ ] ~./libfuzzer_stb_image_concolic/Makefile.toml~ (fuzzer_unix step only contain `cargo build`)
- [x] ./baby_fuzzer_swap_differential/Makefile.toml
- [x] ./qemu_arm_launcher/Makefile.toml
- [x] ./fuzzbench_fork_qemu/Makefile.toml
- [x]  ./qemu_launcher/Makefile.toml
- [ ] ~./nyx_libxml2_standalone/Makefile.toml~ (tasks.test is empty)
- [x] ./fuzzbench/Makefile.toml
- [x] ./frida_libpng/Makefile.toml
- [x] ./fuzzbench_text/Makefile.toml
- [ ] ~./baby_no_std/Makefile.toml~ (`cargo run -Zbuild-std=core,alloc --target x86_64-unknown-linux-gnu > fuzz.log` command didnt store output to fuzz.log )

fix #1037

### CODE REVIEW

**Comment by tokatoka** [comment]
Hi, thanks for this.
i think it's better to do this without changing the fuzzer lib.rs
(because then we can't test a fuzzer that does not write corpus into disk)

You can simply parse the stdout/stderr and search for "corpus: num" from the broker output

File: fuzzers/libfuzzer_libpng_cmin/src/lib.rs [hunk 1]
```diff
@@ -111,7 +114,7 @@ fn fuzz(corpus_dirs: &[PathBuf], objective_dir: PathBuf, broker_port: u16) -> Re
             // RNG
             StdRand::with_seed(current_nanos()),
             // Corpus that will be evolved, we keep it in memory for performance
-            InMemoryCorpus::new(),
+            CachedOnDiskCorpus::new(testcases_dir, 64).unwrap(),
```
Comment (domenukk) [inline fuzzers/libfuzzer_libpng_cmin/src/lib.rs:117]: I wouldn't kick out the InMemoryCorpus personally. It should be enough to check the objective corpus in the end, or maybe parse the logs, as you suggested

**Comment by domenukk** [comment]
this is cool! <3

**Comment by tokatoka** [comment]
You don't need to work on frida_gdi & tinyinst one
They work only on windows and I already added their test

**Comment by SpaceWhite** [comment]
Sorry for the wait. (Quiet busy with personal stuff recently...) This PR is ready for review.

**Comment by tokatoka** [comment]
>  ./qemu_launcher/Makefile.toml has no corpus out directory or stdout log

It puts stdout log for me.

Doesn't it work for you? Maybe you could check if your `cc` points to `gcc`. It doesn't work if it is `clang` afaik

**Comment by SpaceWhite** [comment]
> Doesn't it work for you? Maybe you could check if your cc points to gcc. It doesn't work if it is clang afaik

It doesn't work for me. I'm using Dockerfile in root directory as my development environment. And It uses gcc as default cc.

Btw I could reproduce stdout by using command below
```
> cargo run --release ./libpng_harness
...
[Stats       #1]  (GLOBAL) run time: 0h-7m-20s, clients: 7, corpus: 14967, objectives: 0, executions: 30820441, exec/sec: 69.93k
                  (CLIENT) corpus: 2492, objectives: 0, executions: 5290098, exec/sec: 12.00k, edges: 2632/2673 (98%)
[Stats       #2]  (GLOBAL) run time: 0h-7m-20s, clients: 7, corpus: 14967, objectives: 0, executions: 30820441, exec/sec: 69.93k
                  (CLIENT) corpus: 2503, objectives: 0, executions: 5226073, exec/sec: 11.86k, edges: 2632/2673 (98%)
[Stats       #3]  (GLOBAL) run time: 0h-7m-20s, clients: 7, corpus: 14967, objectives: 0, executions: 30820441, exec/sec: 69.93k
                  (CLIENT) corpus: 2488, objectives: 0, executions: 4860842, exec/sec: 11.03k, edges: 2632/2673 (98%)
[Stats       #5]  (GLOBAL) run time: 0h-7m-20s, clients: 7, corpus: 14967, objectives: 0, executions: 30820441, exec/sec: 69.93k
                  (CLIENT) corpus: 2500, objectives: 0, executions: 5144057, exec/sec: 11.67k, edges: 2632/2673 (98%)
[Stats       #4]  (GLOBAL) run time: 0h-7m-20s, clients: 7, corpus: 14967, objectives: 0, executions: 30820441, exec/sec: 69.93k
                  (CLIENT) corpus: 2485, objectives: 0, executions: 5171265, exec/sec: 11.73k, edges: 2632/2673 (98%)
...
```

But not when timeout command is used
```
 timeout 5s cargo run --release ./libpng_harness 
   Compiling libafl_qemu_sys v0.9.0 (/workspaces/LibAFL/libafl_qemu/libafl_qemu_sys)
warning: CROSS_CC is not set, default to cc (things can go wrong if the selected cpu target (x86_64) is not the host arch (x86_64))
   Compiling libafl_qemu v0.9.0 (/workspaces/LibAFL/libafl_qemu)
root@157cb61fb3d1:/workspaces/LibAFL/fuzzers/qemu_launcher#
```

Any idea why timeout command doesn't produce std output?

**Comment by tokatoka** [comment]
> Any idea why timeout command doesn't produce std output?

because it builds qemu_launcher everytime you run cargo run and it takes more than 5 sec

**Pull request merged**